Deploy failure

3s
Run cloudflare/wrangler-action@v3
🔍 Checking for existing Wrangler installation

🚀 Running Wrangler Commands
  /opt/hostedtoolcache/node/20.20.0/x64/bin/npx wrangler deploy
  
  Cloudflare collects anonymous telemetry about your usage of Wrangler. Learn more at https://github.com/cloudflare/workers-sdk/tree/main/packages/wrangler/telemetry.md
  
   ⛅️ wrangler 3.114.17 (update available 4.65.0)
  -----------------------------------------------
  
  ▲ [WARNING] The version of Wrangler you are using is now out-of-date.
  
    Please update to the latest version to prevent critical errors.
    Run `npm install --save-dev wrangler@4` to update to the latest version.
    After installation, run Wrangler with `npx wrangler`.
  
  
  Total Upload: 61.07 KiB / gzip: 14.97 KiB
  Your worker has access to the following bindings:
  - KV Namespaces:
    - LINKS: $KV_NAMESPACE_ID
  
  ✘ [ERROR] A request to the Cloudflare API (/accounts/***/workers/scripts/gitly-sh/versions) failed.
  
    KV namespace '$KV_NAMESPACE_ID' is not valid.  [code: 10042]
    
    If you think this is a bug, please open an issue at: https://github.com/cloudflare/workers-sdk/issues/new/choose
  
  
  🪵  Logs were written to "/home/runner/.config/.wrangler/logs/wrangler-2026-02-15_03-46-17_004.log"
Error: The process '/opt/hosted

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -42,5 +42,5 @@ jobs:
           apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
           accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
           workingDirectory: apps/worker
-        env:
-          KV_NAMESPACE_ID: ${{ secrets.KV_NAMESPACE_ID }}
+          preCommands: |
+            sed -i "s/\$KV_NAMESPACE_ID/${{ secrets.KV_NAMESPACE_ID }}/g" wrangler.toml

